applications/FluidDynamicsApplication/python_scripts/apply_inlet_process.py

```diff
@@ -18,16 +18,20 @@ def __init__(self, Model, settings):
             "variable_name"   : "VELOCITY",
             "modulus"         : 0.0,
             "constrained"     : true,
-            "direction"       : "automatic_inwards_normal",
+            "direction"       : [1.0,0.0,0.0],
             "interval"        : [0.0,"End"]
         }
         """)
 
         # Trick: allow "modulus" and "direction" to be a double or a string value (otherwise the ValidateAndAssignDefaults might fail)
-        if(settings.Has("modulus")):
-            if(settings["modulus"].IsString()):
+        if (settings.Has("modulus")):
+            if (settings["modulus"].IsString()):
                 default_settings["modulus"].SetString("0.0")
```

Comment:
I agree beforehand it is a hacky solution, however i don't see how we could handle othewise.

One possibility would be to still do the same but to hide it in a function.
For example let's suppose that "modulus" can be either a string or a number (as i the case here)

we can define the functions

* AcceptAsString
* AcceptaAsNumber

and call them, prior to validate, as 

defaults.AcceptAsString("modulus", settings)
defaults.AcceptAsNumber("modulus", settings)

which would still change the defaults under the hood but would hide it.